backend/clubs/views.py

```diff
@@ -4621,42 +4624,47 @@ def cart(self, request, *args, **kwargs):
 
         cart, _ = Cart.objects.get_or_create(owner=self.request.user)
 
-        # this flag is true when a validate operation fails to
-        # replace a ticket. return 200 if true, otherwise 204
-        sold_out_flag = False
-
-        for ticket in cart.tickets.all():
-            # if ticket in cart has been bought, try to replace
-            if ticket.owner or ticket.holder:
-                # lock new ticket until transaction is completed
-                new_ticket = (
-                    Ticket.objects.select_for_update(skip_locked=True)
-                    .filter(
-                        event=ticket.event,
-                        type=ticket.type,
-                        owner__isnull=True,
-                        holder__isnull=True,
-                    )
-                    .first()
+        # Replace in-cart tickets that have been bought/held
+        tickets_to_replace = cart.tickets.filter(
+            owner__isnull=False
+        ) | cart.tickets.filter(holder__isnull=False, holder__ne=self.request.user)
```

Comment:
@aviupadhyayula if you want or behaviour within the ORM...you might be better served with t[`Q` objects](https://docs.djangoproject.com/en/5.0/topics/db/queries/#complex-lookups-with-q-objects)

Reply:
Good catch. Should be fixed.